data/transactions/logic/crypto_test.go

```diff
@@ -117,6 +117,72 @@ byte 0x98D2C31612EA500279B6753E5F6E780CA63EBA8274049664DAD66A2565ED1D2A
 	testAccepts(t, progText, 1)
 }
 
+func TestMimc(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	// Test vectors from https://github.com/giuliop/test-mimc-opcodes/blob/main/testcases/circuit.go
+	// generated by instantiating a zk-circuit that processes MiMC hash of the preimages.
+	// We test success for 32-byte and 96-byte preimages, and failure for preimage input size of 0,
+	// input size not multiple of 32 bytes, and chunks representing values greater than the modulus.
+
+	preImageTestVectors := []string{
+		// FAIL: zero-length
+		"0x",
+		// SUCCEED: 32 bytes, less than modulus
```

Comment:
Could we use a structure more like:
```
testVectors := []struct{input string, output string}{
    {"0x", ""},
    {"0x23....35", "0x12...53"},
}
```
That is, put the input and output (with "" indicating failure) all together so we can see at the site of the test whether it should succeed or fail, rather than hope the different arrays are properly synchronized with the SUCCEED / FAIL comments.

See TestSumhash for somethign similar (though it doesn't have to deal with failures).